security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go

```diff
@@ -102,7 +93,8 @@ func (p Plugin) ExchangeToken(ctx context.Context, trustDomain, k8sSAjwt string)
 	body, _ := ioutil.ReadAll(resp.Body)
 	respData := &federatedTokenResponse{}
 	if err := json.Unmarshal(body, respData); err != nil {
-		stsClientLog.Errorf("Failed to unmarshal response data: %v", err)
+		stsClientLog.Errorf("Failed to unmarshal response data: (HTTP status %d) %v",
+			resp.StatusCode, err)
```

Comment:
Just curious - can resp be nil?

Reply:
Good point, if err is nil, resp is not. https://golang.org/pkg/net/http/#Client.Do

On this regard, we may should change line 98 to avoid dereferencing resp when err is not nil.

Reply:
Yeah, that's the old code there before this PR, could change in follow up.